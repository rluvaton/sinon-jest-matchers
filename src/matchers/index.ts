export { sinonToBeCalled } from './sinonToBeCalled';
export { sinonToBeCalledTimes } from './sinonToBeCalledTimes';
export { sinonToBeCalledWith } from './sinonToBeCalledWith';
export { sinonLastCalledWith } from './sinonLastCalledWith';
export { sinonLastReturnedWith } from './sinonLastReturnedWith';
export { sinonNthCalledWith } from './sinonNthCalledWith';
export { sinonToReturn } from './sinonToReturn';
export { sinonToReturnTimes } from './sinonToReturnTimes';
export { sinonToReturnWith } from './sinonToReturnWith';
