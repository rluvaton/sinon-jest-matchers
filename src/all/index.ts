import * as matchers from '../matchers';

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
const jestExpect = global.expect;

if (jestExpect !== undefined) {
  jestExpect.extend(matchers);
} else {
  throw new Error(
    "Unable to find Jest's global expect. " +
      'Please check you have added sinon-jest-matchers correctly to your jest configuration. ' +
      'See https://github.com/rluvaton/sinon-jest-matchers#setup for help.',
  );
}
