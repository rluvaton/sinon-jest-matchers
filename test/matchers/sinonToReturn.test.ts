import sinon from 'sinon';
import * as matchers from '../../src';

expect.extend(matchers);

describe('sinonToReturn', () => {
  test('.not works only on sinon.spy', () => {
    const fn = function fn() {};

    expect(() => expect(fn).not.sinonToReturn()).toThrowErrorMatchingSnapshot();
  });

  test('passes when returned', () => {
    const fn = sinon.spy(() => 42);
    fn();
    expect(fn).sinonToReturn();
    expect(() => expect(fn).not.sinonToReturn()).toThrowErrorMatchingSnapshot();
  });

  test('passes when undefined is returned', () => {
    const fn = sinon.spy(() => undefined);
    fn();
    expect(fn).sinonToReturn();
    expect(() => expect(fn).not.sinonToReturn()).toThrowErrorMatchingSnapshot();
  });

  test('passes when at least one call does not throw', () => {
    const fn = sinon.spy(causeError => {
      if (causeError) {
        throw new Error('Error!');
      }

      return 42;
    });

    fn(false);

    try {
      fn(true);
    } catch {
      // ignore error
    }

    fn(false);

    expect(fn).sinonToReturn();
    expect(() => expect(fn).not.sinonToReturn()).toThrowErrorMatchingSnapshot();
  });

  test('.not passes when not returned', () => {
    const fn = sinon.spy();

    expect(fn).not.sinonToReturn();
    expect(() => expect(fn).sinonToReturn()).toThrowErrorMatchingSnapshot();
  });

  test('.not passes when all calls throw', () => {
    const fn = sinon.spy(() => {
      throw new Error('Error!');
    });

    try {
      fn();
    } catch {
      // ignore error
    }

    try {
      fn();
    } catch {
      // ignore error
    }

    expect(fn).not.sinonToReturn();
    expect(() => expect(fn).sinonToReturn()).toThrowErrorMatchingSnapshot();
  });

  // Bug in sinon - https://github.com/sinonjs/sinon/issues/2471
  test.skip('.not passes when a call throws undefined', () => {
    const fn = sinon.spy(() => {
      // eslint-disable-next-line no-throw-literal
      throw undefined;
    });

    try {
      fn();
    } catch {
      // ignore error
    }

    expect(fn).not.sinonToReturn();
    expect(() => expect(fn).sinonToReturn()).toThrowErrorMatchingSnapshot();
  });

  test('fails with any argument passed', () => {
    const fn = sinon.spy();

    fn();
    // @ts-expect-error we test that it throws on argument
    expect(() => expect(fn).sinonToReturn(555)).toThrowErrorMatchingSnapshot();
  });

  test('.not fails with any argument passed', () => {
    const fn = sinon.spy();

    // @ts-expect-error we test that it throws on argument
    expect(() => expect(fn).not.sinonToReturn(555)).toThrowErrorMatchingSnapshot();
  });

  test('includes the custom mock name in the error message', () => {
    const fn = sinon.spy(() => 42);
    fn.named('named-mock');
    fn();
    expect(fn).sinonToReturn();
    expect(() => expect(fn).not.sinonToReturn()).toThrowErrorMatchingSnapshot();
  });
});
