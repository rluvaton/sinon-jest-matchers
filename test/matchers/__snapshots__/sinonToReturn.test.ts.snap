// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sinonToReturn .not fails with any argument passed 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22mnot[2m.[22msinonToReturn[2m()[22m

[1mMatcher error[22m: this matcher must not have an expected argument

Expected has type:  number
Expected has value: [32m555[39m"
`;

exports[`sinonToReturn .not passes when all calls throw 1`] = `
"[2mexpect.[22msinonToReturn[2m()[22m

Expected number of returns: >= [32m1[39m
Received number of returns:    [31m0[39m
Received number of calls:      [31m2[39m"
`;

exports[`sinonToReturn .not passes when not returned 1`] = `
"[2mexpect.[22msinonToReturn[2m()[22m

Expected number of returns: >= [32m1[39m
Received number of returns:    [31m0[39m"
`;

exports[`sinonToReturn .not works only on sinon.spy 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22mnot[2m.[22msinonToReturn[2m()[22m

[1mMatcher error[22m: [31mreceived[39m value must be a stub or spy function

Received has type:  function
Received has value: [31m[Function fn][39m"
`;

exports[`sinonToReturn fails with any argument passed 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22msinonToReturn[2m()[22m

[1mMatcher error[22m: this matcher must not have an expected argument

Expected has type:  number
Expected has value: [32m555[39m"
`;

exports[`sinonToReturn includes the custom mock name in the error message 1`] = `
"[2mexpect([22m[31mnamed-mock[39m[2m).[22mnot[2m.[22msinonToReturn[2m()[22m

Expected number of returns: [32m0[39m
Received number of returns: [31m1[39m

1: [31m42[39m"
`;

exports[`sinonToReturn passes when at least one call does not throw 1`] = `
"[2mexpect.[22mnot[2m.[22msinonToReturn[2m()[22m

Expected number of returns: [32m0[39m
Received number of returns: [31m2[39m

1: [31m42[39m
3: [31m42[39m

Received number of calls:   [31m3[39m"
`;

exports[`sinonToReturn passes when returned 1`] = `
"[2mexpect.[22mnot[2m.[22msinonToReturn[2m()[22m

Expected number of returns: [32m0[39m
Received number of returns: [31m1[39m

1: [31m42[39m"
`;

exports[`sinonToReturn passes when undefined is returned 1`] = `
"[2mexpect.[22mnot[2m.[22msinonToReturn[2m()[22m

Expected number of returns: [32m0[39m
Received number of returns: [31m1[39m

1: [31mundefined[39m"
`;
